File: xor/xor.js
_X = [
    [0, 0],
    [0, 1],
    [1, 0],
    [1, 1]
];
_Y = [0, 1, 1, 0];

tf.loadModel("http://localhost:8000/models/js/xor.json").then((model) => {
    model.predict(tf.tensor2d([[1, 1], [0, 0], [0, 1], [1, 0]])).print();
}).catch((resp) => {
    console.log("failed");   
    const model = tf.sequential();
    model.add(tf.layers.dense({units: 16, inputShape: [2], activation: 'relu'}));
    model.add(tf.layers.dense({units: 16, inputShape: [4], activation: 'softmax'}));
    model.add(tf.layers.dense({units: 1, inputShape: [4]}));

    model.compile({loss: 'meanSquaredError', optimizer: tf.train.adam(0.02)});

    const xs = tf.tensor2d(_X);
    const ys = tf.tensor1d(_Y);
    model.fit(xs, ys, {epochs: 100, shuffle: true}).then(() => {
        model.save("downloads://xor");
        model.predict(tf.tensor2d([[1, 1], [0, 0], [0, 1], [1, 0]])).print();
    });
});

File: xor/xor.py
import tflearn
from tflearn.layers.conv import conv_2d, max_pool_2d
from tflearn.layers.core import input_data, dropout, fully_connected
from tflearn.layers.estimator import regression
import tflearn.datasets.mnist as mnist
from random import randint
import tensorflow as tf
import numpy as np;


template_x = [ [0, 0], [0, 1], [1, 0], [1, 1] ]
template_y = [ [0], [1], [1], [0] ]
inputs = np.array(template_x)
targets = np.array(template_y)

input_layer = input_data(shape=[None, 2])
convnet = fully_connected(input_layer, 16, activation='tanh')
#convnet = dropout(convnet, 0.8)
convnet = fully_connected(convnet, 1, activation='tanh')
convnet = regression(convnet, optimizer='adam', learning_rate=0.01, loss='mean_square')

model = tflearn.DNN(convnet)
model.fit(inputs, targets, n_epoch=1000, show_metric=True, batch_size=100, shuffle=True, run_id='xor')
#model.fit(template_x, template_y)
#model.save('models/tflearncnn.model')
#model.load('models/tflearncnn.model');

print(model.predict( [ [1, 1], [0, 0], [1, 0], [0, 1] ] ))